### PR TITLE
feat: Add `spacing` option to `PhpdocAlignFixer`

### PR DESCRIPTION
**Pull request by kusab85**
The ``spacing`` option controls spacing between tag, hint, variable, signature, comment, etc. 
It allows to set same spacing for all tags using a positive integer or different spacings for different tags using an associative array of positive integers ``['tagA' => spacingForA, 'tagB' =>spacingForB]``.

To meet Laravel phpdoc alignment code style criteria configure fixer with:
``['align' => 'left', 'spacing' => ['param' => 2]]``

### CODE REVIEW

**Comment by Wirone** [comment]
@kusab85 please rebase the branch or add me to `gdl2studio` org so I can do it and push refreshed code.

**Comment by kusab85** [comment]
Will do in a near future, @Wirone

**Comment by kusab85** [comment]
Hey @Wirone! 
All checks are green! 
Do the review and merge!

File: src/Fixer/Phpdoc/PhpdocAlignFixer.php
Comment (Wirone) [inline src/Fixer/Phpdoc/PhpdocAlignFixer.php:255]: Minor: can we use heredoc instead of concatenated string? Or maybe single string like for `tags` (even if it's longer than suggested line width)?

**Review by Wirone — CHANGES_REQUESTED** [review]
In general looks good. Personally I wouldn't use it, but I know that Laravel has its own standard with 2 spaces so it will allow them to re-use this fixer with custom config rather than cloning whole class on their side.

**What I miss here** is ability to define default spacing and then override it only for specific tags. For example: 2 spaces by default, but only 1 space for `@return`. With current approach (`int` vs `int[]`) we rather don't have option different than handling some "magic" key in the config array, e.g. `_default`. What do you think?

File: src/Fixer/Phpdoc/PhpdocAlignFixer.php
Comment (kusab85) [reply src/Fixer/Phpdoc/PhpdocAlignFixer.php]: Applied.

File: src/Fixer/Phpdoc/PhpdocAlignFixer.php
Comment (kusab85) [reply src/Fixer/Phpdoc/PhpdocAlignFixer.php]: Single string applied.

**Comment by kusab85** [comment]
Hi @Wirone, I've implemented configurability to ``default spacing`` as you suggested, but some checks failed due to some testing environment issues. Dunno why :-/ I can't fix it.

**Comment by Wirone** [comment]
@kusab85 error for PHP 8.4 is not related to your PR, same problem occured in @mvorisek's PR and I believe it's something that should be fixed separately and merged into master to unlock PRs.

**Comment by mvorisek** [comment]
> @kusab85 error for PHP 8.4 is not related to your PR, same problem occured in @mvorisek's PR and I believe it's something that should be fixed separately and merged into master to unlock PRs.

cherry pick https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7679/commits/28baa2d75eda2db5b6361bd68afc149cb285b119 into master

**Comment by Wirone** [comment]
@mvorisek @kusab85 #7702, but I can't merge my own PR without approval, we need to wait for any other maintainer.

File: src/Fixer/Phpdoc/PhpdocAlignFixer.php
Comment (Wirone) [inline src/Fixer/Phpdoc/PhpdocAlignFixer.php:395]: I believe we can go with:

```suggestion
            ($this->spacing[$tag] ?? $this->spacing[self::DEFAULT_SPACING_KEY] ?? self::DEFAULT_SPACING);
```

and get rid of `PhpdocAlignFixer::$defaultSpacing` property and assignment in the constructor.

**Review by Wirone — CHANGES_REQUESTED** [review]
Minor improvement left, then IMHO it's good to go 👍.

File: src/Fixer/Phpdoc/PhpdocAlignFixer.php
Comment (kusab85) [reply src/Fixer/Phpdoc/PhpdocAlignFixer.php]: The property has been thrown out!

**Review by Wirone — APPROVED** [review]

**Comment by Wirone** [comment]
Thank you @kusab85 🍻!

**Comment by kusab85** [comment]
> Thank you @kusab85 🍻!

You're welcome! Feels nice to contribute ;-)

**Pull request merged**
